libcaf_net/caf/net/packet_writer_impl.hpp

```diff
@@ -18,16 +18,18 @@
 
 #pragma once
 
+#include "caf/net/packet_writer.hpp"
+
 #include "caf/byte.hpp"
 #include "caf/span.hpp"
 
 namespace caf {
 namespace net {
 
 /// Implements the interface for transport and application policies and
-/// dispatches member functions either to `decorator` or `parent`.
+/// dispatches member functions either to `object` or `parent`.
 template <class Object, class Parent>
-class write_packet_decorator {
+class packet_writer_impl final : public packet_writer {
```

Comment:
```suggestion
class packet_writer_decorator final : public packet_writer {
```

I think the `_impl` suffix here is misleading. There exist multiple implementations of `packet_writer`. This one simply decorates another `packet_writer`, so I think the `_decorator` suffix gives a better intuition.